algorithms.go

```diff
@@ -284,6 +279,7 @@ func leakyBucket(ctx context.Context, s Store, c Cache, r *RateLimitReq) (resp *
 	if ok {
 		if item.Value == nil {
 			msgPart := "leakyBucket: Invalid cache item; Value is nil"
+			span := trace.SpanFromContext(ctx)
```

Comment:
What do you think about doing this instead of a local variable. BTW, I like doing this instead of setting the span at the top of each method.
```suggestion
			trace.SpanFromContext(ctx).AddEvent(msgPart, trace.WithAttributes(
				attribute.String("hashKey", hashKey),
				attribute.String("key", r.UniqueKey),
				attribute.String("name", r.Name),
			))
```